creator-node/src/snapbackSM/peerSetManager.js

```diff
@@ -0,0 +1,228 @@
+        throw new Error('getNodeUsers() Error: Unexpected response format during getAllNodeUsers() or getNodePrimaryUsers() call')
+      }
+    })
+
+    return nodeUsers
+  }
+
+  /**
+   * Retrieve users with this node as replica (primary or secondary)
+   *  - Makes single request to discovery node to retrieve all users
+   *
+   * @notice This function depends on a new discprov route and cannot be consumed until every discprov exposes that route
+   *    It will throw if the route doesn't exist
+   * @returns {Object[]} array of objects
+   *  - Each object has schema { primary, secondary1, secondary2, user_id, wallet }
+   */
+  async getAllNodeUsers () {
+    // Fetch discovery node currently connected to libs as this can change
+    if (!this.discoveryProviderEndpoint) {
+      throw new Error('No discovery provider currently selected, exiting')
+    }
+
+    // Request all users that have this node as a replica (either primary or secondary)
+    const requestParams = {
+      method: 'get',
+      baseURL: this.discoveryProviderEndpoint,
+      url: `v1/full/users/content_node/all`,
+      params: {
+        creator_node_endpoint: this.creatorNodeEndpoint
+      }
+    }
+
+    // Will throw error on non-200 response
+    let allNodeUsers
+    try {
+      const resp = await axios(requestParams)
+      allNodeUsers = resp.data.data
+    } catch (e) {
+      throw new Error(`getAllNodeUsers() Error: ${e.toString()}`)
+    }
+
+    return allNodeUsers
+  }
+
+  /**
+   * Retrieve users with this node as primary
+   * Leaving this function in until all discovery providers update to new version and expose new `/users/content_node/all` route
+   * @returns {Object[]} array of objects
+   *  - Each object has schema { primary, secondary1, secondary2, user_id, wallet }
+   */
+  async getNodePrimaryUsers () {
+    // Fetch discovery node currently connected to libs as this can change
+    if (!this.discoveryProviderEndpoint) {
+      throw new Error('No discovery provider currently selected, exiting')
+    }
+
+    const requestParams = {
+      method: 'get',
+      baseURL: this.discoveryProviderEndpoint,
+      url: `users/creator_node`,
+      params: {
+        creator_node_endpoint: this.creatorNodeEndpoint
+      }
+    }
+
+    // Will throw error on non-200 response
+    let nodePrimaryUsers
+    try {
+      const resp = await axios(requestParams)
+      nodePrimaryUsers = resp.data.data
+    } catch (e) {
+      throw new Error(`getNodePrimaryUsers() Error: ${e.toString()}`)
+    }
+
+    return nodePrimaryUsers
+  }
+
+  /**
+   * Select chunk of users to process in this run
+   *  - User is selected if (user_id % this.moduloBase = currentModuloSlice)
+   * @param {Object[]} nodeUsers array of objects of schema { primary, secondary1, secondary2, user_id, wallet }
+   */
+  sliceUsers (nodeUsers) {
+    return nodeUsers.filter(nodeUser =>
+      nodeUser.user_id % this.moduloBase === this.currentModuloSlice
```

Comment:
It would be better to pass in current module slice as opposed to tracking it in two files and having to sync the values

Reply:
Same with module base